frontend/src/metabase-lib/types.ts

```diff
@@ -156,6 +156,12 @@ export type ExternalOp = {
   args: [ColumnMetadata, ...ExpressionArg[]];
 };
 
+export type FilterParts = {
+  operator: FilterOperator;
+  options: Record<string, unknown>;
+  args: [ColumnWithOperators, ...ExpressionArg[]];
+};
+
```

Comment:
Could we please make `column` its own property on the `FilterParts` object and leave `args` to `ExpressionArg[]`? The current approach would most likely lead to boilerplate — many filtering components would need a column without the args, and they'd need to extract it every single time

Reply:
I think I can do that with filter parts, yes.

Reply:
@metamben I'm not sure if we need the `uuid` in `options` for filter parts, would it make sense to omit it?

Reply:
How do you use `options`? Does it need to be sanitized?

Reply:
Will unexpected properties break these flows?

Reply:
I'm not sure yet. I can imagine a need to check if a filter has options, and I can see the FE needing to omit `uuid` in this case. I think we'd safer without the `uuid` here

Reply:
Can you defined either a black or a white list? I doubt it's only `uuid` that can cause trouble...

Reply:
Just went through the code to double-check and I believe we only have two filter options: `case-sensitive` and `include-current`

Reply:
Done. I'm a bit afraid that this might lead to surprises in the future, but they should not be difficult to spot/fix.